### ISSUE TITLE
Review timezone definition

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#1103]
This is a placeholder, based on

> `toLocaleString()` is not working as expected because the date returned by fractal-sever doesn't follow [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) standard, which requires to put a `Z` at the end of the string to represent a date in UTC. I think we should fix this on fractal-server side.
> 
> On my browser:
> 
> ```javascript
> new Date('2023-12-20T09:49:27.914030').toLocaleString() // "12/20/2023, 9:49:27 AM"
> new Date('2023-12-20T09:49:27.914030Z').toLocaleString() // "12/20/2023, 10:49:27 AM"
> ```
> 
> _Originally posted by @zonia3000 in https://github.com/fractal-analytics-platform/fractal-web/issues/229#issuecomment-1864211315_

**Comment by tcompa** [comment]
For the record, this minimal-working example seems to work - if we assume that we have UTC timestamps from the DB:
```python
from pydantic import BaseModel, validator  # type: ignore
from datetime import datetime, timezone
from devtools import debug  # type: ignore


def convert_datetime_to_iso_8601_with_z_suffix(dt: datetime) -> str:
    return dt.isoformat().replace("+00:00", "Z")


class JobRead2(BaseModel):
    timestamp: datetime
    _normalize_datetimes = validator("timestamp", allow_reuse=True)(convert_datetime_to_iso_8601_with_z_suffix)

    class Config:
        json_encoders = {
            datetime: convert_datetime_to_iso_8601_with_z_suffix,
        }

now = datetime.now(tz=timezone.utc)
job = JobRead2(timestamp=now)
debug(job)
# x.py:21 <module>
#     job: JobRead2(
#         timestamp='2023-12-20T11:57:17.666294Z',
#     ) (JobRead2)
```

However, we (with @ychiucco) are still not sure about the specific point where fastapi performs serialization (and through which method). Once we understand this, the solution could be something like the snippet above.

**Comment by tcompa** [comment]
Note that this is to be checked both in `Read` models and in the query parameters of the admin endpoints.

**Comment by tcompa** [comment]
Within the CI, at least for the current GET-jobs and GET-project endpoints, the response includes timestamp that end with `+00:00` - see commit a4a37d190c7ea2af804298b8b6afe9b8f72c63df by @ychiucco.

To be verified: where did the no-time-zone timestamps mentioned in https://github.com/fractal-analytics-platform/fractal-web/issues/229#issuecomment-1865805065 appear? (cc @zonia3000)

**Comment by zonia3000** [comment]
> where did the no-time-zone timestamps mentioned in [fractal-analytics-platform/fractal-web#229 (comment)](https://github.com/fractal-analytics-platform/fractal-web/issues/229#issuecomment-1865805065) appear? (cc @zonia3000)

Using fractal-server 1.4.1a2:

* `/api/v1/project/1/job`
* `/admin/job`

**Comment by tcompa** [comment]
Thanks Sonia.

It looks like we have some unexpected behavior that depends on where we run the CI, when using sqlite.

The relevant test (that checks the presence of `+00:00` at the end of the timestamp) passes with this config:
- macos
- sqlite3 v3.39.5

but it fails (with the traceback below) in GitHub CI or in my local config:
- ubuntu 22.04
- sqlite3 v3.37.2

```python traceback
>           assert res.json()["timestamp_created"].endswith("+00:00")
E           AssertionError: assert False
E            +  where False = <built-in method endswith of str object at 0x7fb1e0874260>('+00:00')
E            +    where <built-in method endswith of str object at 0x7fb1e0874260> = '2023-12-21T12:01:28.062053'.endswith
```

Note (in case it matters) that my locale is
```
$ locale

LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=en_GB.UTF-8
LC_TIME=en_GB.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=en_GB.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=en_GB.UTF-8
LC_NAME=en_GB.UTF-8
LC_ADDRESS=en_GB.UTF-8
LC_TELEPHONE=en_GB.UTF-8
LC_MEASUREMENT=en_GB.UTF-8
LC_IDENTIFICATION=en_GB.UTF-8
LC_ALL=
```


----

Based on sqlite3 release notes, I see the following possibly related entries:

> 3.38.0:
> Enhancements to [date and time functions](https://www.sqlite.org/lang_datefunc.html):
>     Added the [unixepoch() function](https://www.sqlite.org/lang_datefunc.html#uepch).
>     Added the [auto modifier](https://www.sqlite.org/lang_datefunc.html#automod) and the [julianday modifier](https://www.sqlite.org/lang_datefunc.html#jdmod).

> 3.38.1
> Fix the [localtime modifier](https://www.sqlite.org/lang_datefunc.html#localtime) of the [date and time functions](https://www.sqlite.org/lang_datefunc.html) so that it preserves fractional seconds.


---

TL;DR
1. Within #1102 I will temporarily skip the assertion, when using sqlite rather than postgres.
2. If it's a sqlite-only issue (to be checked), let's re-assess what to do. It'd be a bit annoying to add a custom serialization procedure that is never actually used in postgres.
3. If there remains some unexpected behavior in the postgres version, then we should definitely come back to this issue.
